internal/api/describe_test.go

```diff
@@ -0,0 +1,156 @@
+package api
+
+import (
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestDescribeFull(test *testing.T) {
+	path := filepath.Join(util.RootDirForTesting(), "api.json")
+
+	var expectedDescription core.APIDescription
+	err := util.JSONFromFile(path, &expectedDescription)
+	if err != nil {
+		test.Fatalf("Failed to load api.json: %v", err)
+	}
+
+	fullDescription := Describe(*GetRoutes())
+
+	if !reflect.DeepEqual(&expectedDescription, fullDescription) {
+		test.Fatalf("Unexpected API Descriptions. Expected: '%s', actual: '%s'.",
+			util.MustToJSONIndent(expectedDescription), util.MustToJSONIndent(fullDescription))
+	}
+}
+
+func TestDescribeEmptyRoutes(test *testing.T) {
+	routes := []core.Route{}
+	description := Describe(routes)
+
+	if len(description.Endpoints) != 0 {
+		test.Errorf("Unexpected number of endpoints. Expected: '0', actual: '%d'.", len(description.Endpoints))
+	}
+}
+
```

Comment:
The tests starting here don't really seem necessary.
I know it sounds strange for someone to say "test less",
but there is a maintenance cost associated with each test so we want to make sure each test has a purpose.

 - TestDescribeNonAPIRoute -- The full routes already has non-api routes.
 - TestDescribeMultipleAPIRoutes -- The full routes already has multiple API routes.
 - TestDescribeDuplicateBasePaths -- This is technically allowed, since they may have different HTTP methods.
 - TestDescribeEmptyRequestAndResponseTypes -- This should not be possible.